includes/class-wc-stripe-customer.php

```diff
@@ -741,6 +746,94 @@ public function get_payment_methods( $payment_method_type ) {
 		return empty( $payment_methods ) ? [] : $payment_methods;
 	}
 
+	/**
+	 * Get all payment methods for a customer.
+	 *
+	 * @param string[] $payment_method_types The payment method types to look for using Stripe method IDs. If the array is empty, it implies all payment method types.
+	 * @param int      $limit                The maximum number of payment methods to return. If the value is -1, no limit is applied.
+	 * @return array
+	 */
+	public function get_all_payment_methods( array $payment_method_types = [], int $limit = -1 ) {
+		if ( ! $this->get_id() ) {
+			return [];
+		}
+
+		$cache_key = self::PAYMENT_METHODS_TRANSIENT_KEY . '__all_' . $this->get_id();
+		$all_payment_methods = get_transient( $cache_key );
```

Comment:
Shouldn't we use the `WC_Stripe_Database_Cache` here?

Reply:
We could, but I was wary of adding new code and adding new data to the database cache before we have a cleanup mechanism (which I created an issue for earlier today in https://github.com/woocommerce/woocommerce-gateway-stripe/issues/4569).

For large stores with regular customers, using the database cache would result in a big jump in the options size, which could easily drive other performance issues on sites.